src/components/LocalNav/LocalNav.tsx

```diff
@@ -1,108 +1,189 @@
-import { Suspense, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { useParams } from 'react-router-dom'
-import { Accordion, AccordionDetails, AccordionSummary, Box, Divider, Stack, Typography } from '@common/components'
-import { ExpandMore } from '@common/icons'
+import { useNavigate, useParams } from 'react-router-dom'
+import { Divider, Drawer, Typography, Grid, List, ListItem, ListItemButton, ListItemText, Skeleton } from '@common/components'
+import { FiberManualRecord } from '@common/icons'
 import {
   SkeletonList,
-  useLearningPathElement as _useLearningPathElement,
-  useLearningPathTopic as _useLearningPathTopic
+  Fraction
 } from '@components'
-import { LearningPathElement, Topic } from '@core'
-import LazyLoadingLearningPathElement from './LazyLoadingLearningPathElement'
+import { Topic } from '@core'
+import { useEffect, useState } from 'react'
+import { Theme } from '@common/theme'
+import { useMediaQuery, useTheme, useLearningPathTopic as _useLearningPathTopic, useLearningPathTopicProgress } from '@common/hooks'
 
+type TopicListItemProps = {
+  topic: Topic;
+  topicProgress: number[];
+  isProgressLoading: boolean;
+  courseId: string;
+  topicId: string;
+}
+
+/**
+ * Topic list item component, displaying a topic with its done learning elements out of total learning elements.
+ * @param
+ * topic: Topic - The topic to display.
+ * index: number - The index of the topic in the topics array.
+ * topicProgress: number[][] - The progress of each topic.
+ * isProgressLoading: boolean - Whether the progress is loading.
+ * courseId: string - The course id.
+ * topicId: string - The topic id.
+ * @returns
+ * A JSX Element with the rendered topic list item.
+ */
+
+const TopicListItem = ({ topic, topicProgress, isProgressLoading, courseId, topicId}: TopicListItemProps) => {
```

Comment:
Gehört in eine eigene File. Könnte man LocalNavItem nennen, dann wäre es vielleicht eindeutiger was damit gemeint ist.